src/judges/codeforces/getContests.js

```diff
@@ -1,46 +1,39 @@
 // @flow
-import request from 'request';
+import request from 'request-promise-native';
 import type { Contest } from '../../types';
 
-export function getContests(): Promise<Array<Contest>> {
-  return new Promise((resolve, reject) => {
-    // making request to the codeforces api
-    request('http://codeforces.com/api/contest.list', (error: Object, response: Object, body: string) => {
-      if (response && response.statusCode == 200) {
-        let contests: Array<Contest> = [];
+export async function getContests(): Promise<Array<Contest>> {
+  const body = await request('http://codeforces.com/api/contest.list');
```

Comment:
knowing codeforces, this is most likely going to break at some point
maybe i am being stupid regarding promises, but shouldn't we deal with it explicitly?

Reply:
If it fails the await will throw an exception and the promise will be rejected.